archunit/src/main/java/com/tngtech/archunit/library/Architectures.java

```diff
@@ -223,6 +222,7 @@ private void checkEmptyLayers(JavaClasses classes, EvaluationResult result) {
         private EvaluationResult evaluateLayersShouldNotBeEmpty(JavaClasses classes, LayerDefinition layerDefinition) {
             return classes().that(layerDefinitions.containsPredicateFor(layerDefinition.name))
                     .should(notBeEmptyFor(layerDefinition))
+                    .allowEmptyShould(true)
```

Comment:
I guess you mean `allowEmptyShould(false)`? (otherwise I don't understand the question, since we're already using `allowEmptyShould(true)` :wink:) So theoretically yes, but the other method is nicer, because it doesn't crash immediately but gather the violations. So with `notBeEmptyFor(layerDefinition)` you will get (possibly multiple) violations like `layer x is empty`. With `allowEmptyShould(false)` you would get one error "rule failed to check any classes" (but which rule? :grimacing: Since there are several nested rules involved). I get it that it's not super nice, but for me at the moment it's "okay", because it's a hidden implementation details encapsulated in one private method. I can add a comment though, maybe that helps to understand it :man_shrugging:
Thinking further about it, maybe it would be nice to add the description to the exception :thinking: Even though it's irrelevant here, but if there are multiple rules nested you would at least have a chance to understand which rule is the problem :thinking:

Reply:
Yes, I meant `allowEmptyShould(false)` 🤦🏼‍♂️

Ah ok, yes, then it makes totally sense for me to use it like it is.